project/Dependencies.scala

```diff
@@ -125,8 +126,9 @@ object Dependencies {
 
   val javaSdkTestKit = deps ++= Seq(
     testContainers,
-    junit4 % Provided,
-    junit5 % Provided,
```

Comment:
This is the most impactful change. 

So far, our testkit was depending directly on a class from junit4. It has junit5 as dependency, but it was not used. 

Users were required to have junit4 in their pom because it was `Provided`. 

It's annoying now to keep them as `Provided` because the testkit now requires both libraries. That means that users will need to add both to their pom.xml.

By making them transitive through the testkit, we ensure that users will have both and can pick the one they prefer. 

On our side, new projects (created from mvn archetype) and new generated code will use junit5.

Reply:
I don't see any point in supporting both junit versions. JUnit 5 was released in 2017. We could support both for a short transition time, but we should be very clear that migration is indispensable.